### ISSUE TITLE
Read base directory and their mappings from config file

### ISSUE THREAD

**Issue by apfong** [UCLA-CS130/Mr.-Robot-et-al.#30]
The config file should specify the base path as well as mappings of URL paths to specific handlers. For example, `/echo` could echo request and `/static` could be used for serving static files.

**Comment by KyCodeHuynh** [comment]
As discussed offline, here is how our new config file format will be:

```
location /path/in/browser {
    root    /path/on/filesystem
    action  request-handler-type # echo
}

location /path/in/browser/another/one {
    root    /path/on/filesystem/subfolder
    action  request-handler-type # serve_static
}

# 404 if not matching a location
# 404 if match location, but file not found 

Initial implementation idea: 

Location struct with Root and Action fields (which are just strings)
```

**Comment by KyCodeHuynh** [comment]
Implementation idea: internal storage is a <unordered_map> of <vector of strings> to a string, where a vector is an absolute path to a property: 

listen -> 8080 // lookup("listen") => "8080"

location /echo -> action -> echo // lookup(["location /echo", "action"]) => "echo"

**Comment by apfong** [comment]
Addressed in [PR #37](https://github.com/UCLA-CS130/Mr.-Robot-et-al./pull/37)
